### PR TITLE
fix best-practices guide changes

### PR DESCRIPTION
**Pull request by mschreml**
new PR after problems in #3373

backport needed for 10.7 and 10.6

### CODE REVIEW

File: docs/best-practices.md [hunk 1]
```diff
@@ -102,23 +102,38 @@ Prefix: `xref:`
 Reference: [`Cross Reference`](https://asciidoctor.org/docs/user-manual/#xref)
 
 In a nutshell, an internal link called `Cross Reference` can link to
+- a in-page reference
```
Comment (EParzefall) [inline docs/best-practices.md:105]:
```suggestion
- an in-page reference
```

File: docs/best-practices.md [hunk 2]
```diff
@@ -102,23 +102,38 @@ Prefix: `xref:`
...
 - a documentation file
-- a section title inside a documentation file
-- a reference to an anchor set inside a documentation file
+- a section-title or anchor-name inside a documentation file
```
Comment (EParzefall) [inline docs/best-practices.md:107]:
```suggestion
- a section title or anchor name inside a documentation file
```

File: docs/best-practices.md [hunk 3]
```diff
@@ -102,23 +102,38 @@ Prefix: `xref:`
...
+All content you want to reference **must be** inside the directory structure of `modules/`.
 
-An `xref` is written in following example style: `xref:module_name:<path>/file.adoc#section[Printed Name]`
+An `xref` is usually written in the following example style:  <br/>
+`xref:module_name:<path>/file.adoc#section[Printed Name]`
 
 Where `module_name:`, `#section` and `[Printed Name]` are optional components.
 `module_name:` is mandatory when referenced content is not in the same module.
-`<path>` is the path to your referenced file.
+`<path>` is the relative path via the module-name to your referenced file.
```
Comment (EParzefall) [inline docs/best-practices.md:116]:
```suggestion
`<path>` is the relative path via the module name to your referenced file.
```

File: docs/best-practices.md [hunk 4]
```diff
@@ -102,23 +102,38 @@ Prefix: `xref:`
...
 
 You can reference a section or an anchor inside the same file, another file - even in another module.
+Section titles automatically create references, where the text is converted to lower characters
```
Comment (EParzefall) [inline docs/best-practices.md:119]:
```suggestion
Section titles automatically create references, where the text is converted to lower case characters 
```

File: docs/best-practices.md [hunk 5]
```diff
@@ -102,23 +102,38 @@ Prefix: `xref:`
...
+
+- This example references another file and a particular section title: <br/>
+`xref:configuration/server/occ_command.adoc#apps-commands[the Market app]`
+
+- This example references another file in another module and a particular section title: <br/>
+`xref:admin_manual:configuration/server/occ_command.adoc#apps-commands[the Market app]`
+
+**Strongly** In general it`s advisable to use a ToC ([Table of Contents](#table-of-contents)) instead of a list of xref´s.
+
+**Note:** Use a broken link checker regularly to find broken links to section-titles or anchors.
```
Comment (EParzefall) [inline docs/best-practices.md:135]:
```suggestion
**Note:** Use a link checker regularly to find broken links to section titles or anchors.
```

**Review by EParzefall — APPROVED** [review]
Fixed some grammar issues.

**Comment by mschreml** [comment]
backported to:

10.7 - #3385 
10.6 - #3386

**Pull request merged**